sql/src/main/java/io/crate/analyze/MultiSourceSelect.java

```diff
@@ -93,12 +93,22 @@ public static MultiSourceSelect createWithPushDown(RelationNormalizer relationNo
         Function<? super Symbol, ? extends Symbol> convertFieldInSymbolsToNewRelations =
             FieldReplacer.bind(convertFieldToPointToNewRelations);
 
+        // remap the sources according to their QualifiedName which might have changed
+        Map<QualifiedName, AnalyzedRelation> newSources = new LinkedHashMap<>();
+        for (Map.Entry<QualifiedName, AnalyzedRelation> source : mss.sources.entrySet()) {
+            AnalyzedRelation analyzedRelation = source.getValue();
+            newSources.put(analyzedRelation.getQualifiedName(), analyzedRelation);
+        }
+
+        Function<QualifiedName, QualifiedName> sourceReplaceFun = qName -> mss.sources().get(qName).getQualifiedName();
         for (JoinPair joinPair : mss.joinPairs) {
             joinPair.replaceCondition(convertFieldInSymbolsToNewRelations);
+            joinPair.replaceSourceNames(sourceReplaceFun);
```

Comment:
I wonder if we should simply create new `joinPairs` given that we're creating a new `MultiSourceSelect` anyway?